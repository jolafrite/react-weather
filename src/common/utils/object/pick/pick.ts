export const pick = <T, K extends keyof T>(obj: T, paths: K[]): Pick<T, K> => {
  return { ...paths.reduce((mem, key) => ({ ...mem, [key]: obj[key] }), {}) } as Pick<T, K>;
}
