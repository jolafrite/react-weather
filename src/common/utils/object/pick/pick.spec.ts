import { pick } from "./pick";

describe('pick', () => {

  test('should return an new array with the picked keys', () => {
    const object = {
      key1: 'value1',
      key2: 'value2',
      key3: 'value3',
    }
    expect(
      pick(object, ['key2', 'key3'])
    ).toMatchSnapshot();

  });

});
