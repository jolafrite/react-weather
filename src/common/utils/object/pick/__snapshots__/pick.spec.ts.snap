// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pick should return an new array with the picked keys 1`] = `
Object {
  "key2": "value2",
  "key3": "value3",
}
`;
